vars/rpmMetaValidation.groovy

```diff
@@ -41,4 +41,34 @@ def call(Map args = [:]) {
         println("Meta data for $key is validated")
     }
     println("Validation for meta data of RPM distribution completed.")
+
+    // Validate the distribution signature
+    def checksig = sh (
+            script: "rpm -K -v $distFile",
+            returnStdout: true
+    ).trim()
+    println("Signature check of the rpm distribution file is: \n" + checksig)
+    def keyList = ["Header V4 RSA/SHA512 Signature, key ID 9310d3fc", "Header SHA256 digest",
+                   "Header SHA1 digest", "Payload SHA256 digest",
+                   "V4 RSA/SHA512 Signature, key ID 9310d3fc", "MD5 digest"]
+    def presentKey = []
+    for (line in checksig.split('\n')) {
+        def key = line.split(':')[0].trim()
+        if (key == distFile) {
+            continue
+        } else {
+            assert line.split(':', 2)[1].trim().contains("OK")
+            println(key + " is validated as: " + line)
+            presentKey.add(key)
+        }
+    }
+    println("Validation all key digests starts: ")
+    for (digest in keyList) {
+        if (presentKey.contains(digest)) {
+            println("Key digest \"$digest\" is validated to be present.")
+        } else {
+            error("Key digest \"$digest\" is not present.")
```

Comment:
Can we convert this to assert like above instead of throwing error?

Reply:
I had `assert` originally but changed to what `if` cause the error log of this logic looks more descriptive and straightforward to me than just using assert. But I will take your suggestion and change it back. :)